rl.py

```diff
@@ -32,144 +33,281 @@
 import random
 import argparse
 import time
+from colorama import Fore
+
+import sofagym
+from sofagym.envs import *
 
 sys.path.insert(0, str(pathlib.Path(__file__).parent.absolute())+"/../")
 sys.path.insert(0, str(pathlib.Path(__file__).parent.absolute()))
 
 
 # Adapted from rl-agents
-def load_environment(id, rank, seed = 0):
+def load_environment(env_name, rank=0, seed=0, max_episode_steps=100):
     def _init():
         __import__('sofagym')
-        env = gym.make(id)
+        env = gym.make(env_name)
         env.seed(seed + rank)
         env.reset()
+        env = gym.wrappers.TimeLimit(env, max_episode_steps=max_episode_steps)
         return env
 
     return _init
 
 
-
-def test(env, model, epoch, n_test=1, render = False):
+def test(env, model, test, n_steps, n_episode=1, render=False):
     if render:
         env.config.update({"render":1})
+
     r, final_r = 0, 0
-    for t in range(n_test):
-        print("Start >> Epoch", epoch, "- Test", t)
+
+    for t in range(n_episode):
+        print("Start >> Epoch", test, "- Episode", t+1)
         obs = env.reset()
+
         if render:
             env.render()
+        
         rewards = []
         done = False
         id = 0
+        
         while not done:
             action, _states = model.predict(obs, deterministic = False)
             obs, reward, done, info = env.step(action)
             if render:
-                print("Test", t, "- Epoch ", id ,"- Took action: ", action, "- Got reward: ", reward)
+                print("Episode", t+1, "- Step ", id+1 ,"- Took action: ", action, "- Current State: ", obs, "- Got reward: ", reward, "- Done: ", done)
                 env.render()
             rewards.append(reward)
-            id+=1
-        print("Done >> Test", t, "- Reward = ", rewards, "- Sum reward:", sum(rewards))
-        r+= sum(rewards)
-        final_r+= reward
+            id +=1
+        
+        print("Done >> Episode", t+1, "- Reward = ", rewards, "- Sum reward:", sum(rewards))
+        r += sum(rewards)
+        final_r += reward
+    
     print("[INFO]  >> Mean reward: ", r/n_test, " - Mean final reward:", final_r/n_test)
+
+    env.close()
+    
     return r/n_test, final_r/n_test
 
+
 def sec_to_hours(seconds):
     a=str(seconds//3600)
     b=str((seconds%3600)//60)
     c=str((seconds%3600)%60)
     d=["{} hours {} mins {} seconds".format(a, b, c)]
     return d
 
+
+results_dir = "./Results_benchmark"
+models_dir = f"{results_dir}/models"
+logs_dir = f"{results_dir}/logs"
+videos_dir = f"{logs_dir}/videos"
+
+envs = {
+        1: 'bubblemotion-v0',
```

Comment:
Why `v0` at the end of all names?

Reply:
It's a convention in gym to add the env version after the name, in this case version 0. If a significant change is made to the environment like changing how its reward is calculated for example, it would be appropriate to change it to v1.